tests/microgrid/test_graph.py

```diff
@@ -1578,3 +1574,48 @@ def test_without_grid_meters(self) -> None:
         assert graph.is_chp(chp)
         assert graph.is_chp_chain(chp_meter)
         assert graph.is_chp_chain(chp)
+
+
+def add_components(graph: gr._MicrogridComponentGraph, *components: Component) -> None:
+    """Add components to the test graph.
+
+    Args:
+        graph: The graph to add the components to.
+        *components: The components to add.
+    """
+    _add_components(graph, *components)
+
+
+def add_connections(
+    graph: gr._MicrogridComponentGraph, *connections: Connection
+) -> None:
+    """Add connections to the test graph.
+
+    Args:
+        graph: The graph to add the connections to.
+        *connections: The connections to add.
+    """
+    graph._graph.add_edges_from(
+        (c.start, c.end, {gr._DATA_KEY: c}) for c in connections
+    )
```

Comment:
I don't understand why we have 2 similar functions  `_add_connections()` and `add_connections()`. Same question for `add_components()`. Can you please clarify the usage of these functions? Is it because `create_test_graph()` needs them?

Reply:
Long story short, I was trying out Cursor to help me with this refactor and it added these functions without me noticing 😬

Note to self: review the diffs before creating the PR. 😁